common/models/user.js

```diff
@@ -604,8 +604,8 @@ module.exports = function(User) {
     };
 
     UserModel.setter.password = function(plain) {
-      if (typeof plain !== 'string') {
-        return;
+      if (typeof plain === 'number') {
+        plain = plain.toString();
       }
```

Comment:
I don't think that this check should be removed entirely. What if someone makes a mistake somewhere and `plain` is of type `object`? Instead, I think that you should add the check for the number above the check for string. 

This could also be cleaned up to one line:
```node
if (typeof plain !== 'string') return;
```